src/analyzer/analyzersilence.cpp

```diff
@@ -117,29 +94,37 @@ void AnalyzerSilence::storeResults(TrackPointer tio) {
     double introStart = mixxx::kAnalysisChannels * m_iSignalStart;
     double outroEnd = mixxx::kAnalysisChannels * m_iSignalEnd;
 
-    if (shouldUpdateMainCue(tio->getCuePoint())) {
-        tio->setCuePoint(CuePosition(introStart, Cue::AUTOMATIC));
+    CuePointer pAudibleSound = pTrack->findCueByType(Cue::Type::AudibleSound);
+    if (pAudibleSound == nullptr) {
+        pAudibleSound = pTrack->createAndAddCue();
+        pAudibleSound->setType(Cue::Type::AudibleSound);
+        pAudibleSound->setPosition(introStart);
+        pAudibleSound->setLength(outroEnd - introStart);
     }
 
-    CuePointer pIntroCue = tio->findCueByType(Cue::INTRO);
-    if (!pIntroCue) {
-        pIntroCue = tio->createAndAddCue();
-        pIntroCue->setType(Cue::INTRO);
-        pIntroCue->setSource(Cue::AUTOMATIC);
+    CuePointer pIntroCue = pTrack->findCueByType(Cue::Type::Intro);
+
+    double mainCue = pTrack->getCuePoint().getPosition();
+    // NOTE: the actual default for this ConfigValue is set in DlgPrefDeck.
+    if (m_pConfig->getValue(ConfigKey("[Controls]", "SetIntroStartAtMainCue"), false)
+            && pIntroCue == nullptr && mainCue != kCueNotSet && mainCue != 0.0) {
+        introStart = mainCue;
+    } else if (mainCue == kCueNotSet) {
```

Comment:
Can we check this first. This will stream line the condition above.  The default cue is at 0:00. 
I think we want to move it in this case too.

Reply:
Good idea. Done: b2d35ef210d77003fa3ea01846dfcc2cfba70242

Reply:
>  The default cue is at 0:00. I think we want to move it in this case too.

I forgot that was intentionally implemented in #1242. Somewhere along the way I broke that. I reimplemented it in b2d8bdd3db0f316505ed2eca215573b29a9cf5f9.